### ISSUE TITLE
Updating News command

### ISSUE THREAD

**Issue by BluePigman** [BluePigman/BluePagmanBot#21]
Google RSS News feed must have changed, as the command now returns long links from google RSS, not the direct url, example: 
https://news.google.com/rss/articles/CBMieGh0dHBzOi8vd3d3Lm5wci5vcmcvc2VjdGlvbnMvc2hvdHMtaGVhbHRoLW5ld3MvMjAyNC8wNy8xNy9ueC1zMS01MDQxOTg1L21hbW1vZ3JhbS1ndWlkZWxpbmVzLWJyZWFzdC1jYW5jZXItb3ZlcnRyZWF0bWVudNIBAA?oc=5&hl=en-US&gl=US&ceid=US:en

**Comment by BluePigman** [comment]
```python
def decode_url(google_news_url):
    code_pattern = r'\/articles\/(.*?)\?'
    code_match = re.search(code_pattern , google_news_url)
    if not code_match:
        return "Error: Not a valid link"

    encoded_url = code_match.group(1)

    # Add the missing padding
    missing_padding = len(encoded_url) % 4
    if missing_padding:
        encoded_url += '=' * (4 - missing_padding)

    decoded_url = str(base64.b64decode(encoded_url), 'utf-8', errors='ignore')
    if not decoded_url:
        return "None"

    url_pattern = r'http.+'
    url_match = re.search(url_pattern , decoded_url)
    print("HMM: ", decoded_url)
    if not url_match:
        return "Error: Missing http:"
        
    decoded_url = url_match.group(0)
    head, _, _ = decoded_url.partition("\x01")
    return head


```
This has a few caveats:

**Comment by BluePigman** [comment]
https://news.google.com/rss/articles/CCAiC3FIckhzeHN1MjlNmAEB?oc=5&hl=en-CA&gl=CA&ceid=CA:en
(Tadej Pogacar Unleashed: Intense Tour de France 2024 Battle - FloBikes) 
gives error Error: Missing http:, the link redirects to youtube

**Comment by BluePigman** [comment]
!news Tadej Pogacar Unleashed

https://news.google.com/rss/articles/CBMiVWh0dHBzOi8vd3d3LmJyZWl0YmFydC5jb20vbmV3cy9wb2dhY2FyLWV4dGVuZHMtdG91ci1kZS1mcmFuY2UtbGVhZC13aXRoLXB5cmVuZWVzLXdpbi_SAVlodHRwczovL3d3dy5icmVpdGJhcnQuY29tL25ld3MvcG9nYWNhci1leHRlbmRzLXRvdXItZGUtZnJhbmNlLWxlYWQtd2l0aC1weXJlbmVlcy13aW4vYW1wLw?oc=5&hl=en-CA&gl=CA&ceid=CA:en
Invalid base64-encoded string: number of data characters (241) cannot be 1 more than a multiple of 4
RwczovL3d3dy5icmVpdGJhcnQuY29tL25ld3MvcG9nYWNhci1leHRlbmRzLXRvdXItZGUtZnJhbmNlLWxlYWQtd2l0aC1weXJlbmVlcy13aW4vYW1wLw?oc=5&hl=en-CA&gl=CA&ceid=CA:en
Invalid base64-encoded string: number of data characters (241) cannot be 1 more than a multiple of 4

**Comment by BluePigman** [comment]
!news drdisrespect

https://news.google.com/rss/articles/CBMidmh0dHBzOi8vd3d3LmRleGVydG8uY29tL3R3aXRjaC9kci1kaXNyZXNwZWN0LWZhbi10aHJvd3MtY29zdHVtZS1pbi10cmFzaC1hZnRlci1zdHJlYW1lci1hZG1pdHMtdG8tZG1pbmctbWlub3ItMjc5ODI5MC_SAQA?oc=5&hl=en-CA&gl=CA&ceid=CA:en
Incorrect padding

**Comment by BluePigman** [comment]
Fixed in [fa5b7d9](https://github.com/BluePigman/BluePagmanBot/commit/fa5b7d9c4e499577035d877d4851a72f8be0c3b8)
 ·
